lib/proxy/router.go

```diff
@@ -229,7 +229,10 @@ func (r *Router) DialHost(ctx context.Context, clientSrcAddr, clientDstAddr net.
 	}
 	span.AddEvent("retrieved target server")
 
-	principals := []string{host}
+	principals := []string{
+		host,
+		host + "." + clusterName,
```

Comment:
Do we want to add a comment here explaining that this particular principal is needed for the leaf clusters use-case?